libs/design-system/src/navigation/NavMenu.tsx

```diff
@@ -1,40 +1,106 @@
-import React, { ReactNode } from 'react';
+import React, { ReactNode, useState } from 'react';
 import { NavLink } from 'react-router-dom';
-import { Group } from '@mantine/core';
+import { Group, Transition, Popover } from '@mantine/core';
 import useStyles from './NavMenu.styles';
+import usePopoverStyles from './NavLinkPopover.styles';
 
 interface INavMenuProps {
   menuItems: {
     icon: ReactNode;
     label: string;
     link: string;
     testId?: string;
-    rightSide?: ReactNode;
+    rightSide?: ReactNode | { component: ReactNode; displayOnHover: boolean };
     condition?: boolean;
+    tooltipLabel?: string;
   }[];
 }
 
 export function NavMenu({ menuItems }: INavMenuProps) {
   const { classes } = useStyles();
+  const { classes: popoverClasses } = usePopoverStyles();
+  const [popoverOpened, setPopoverOpened] = useState(false);
+
+  const [isHovered, setIsHovered] = useState<string>(null);
 
   return (
     <div>
       {menuItems
         .filter(({ condition = true }) => condition)
-        .map(({ icon, link, label, testId, rightSide }) => (
-          <NavLink
-            key={link}
-            to={link}
-            className={({ isActive }) => (isActive ? classes.linkActive : classes.link)}
-            data-test-id={testId}
-          >
-            <Group spacing={10}>
-              <div className={classes.linkIcon}>{icon}</div>
-              <div className={classes.linkLabel}>{label}</div>
-            </Group>
-            <div>{rightSide}</div>
-          </NavLink>
-        ))}
+        .map(({ icon, link, label, testId, rightSide, tooltipLabel }) => {
+          const withOnHover = (rightSide as { component: ReactNode; displayOnHover: boolean })?.displayOnHover || false;
+
+          return (
+            <NavLink
+              key={link}
+              to={link}
+              className={({ isActive }) => (isActive ? classes.linkActive : classes.link)}
+              data-test-id={testId}
+              onMouseEnter={() => setIsHovered(link)}
+              onMouseLeave={() => setIsHovered(null)}
+            >
+              <Group spacing={10}>
+                <div className={classes.linkIcon}>{icon}</div>
+                <div className={classes.linkLabel}>{label}</div>
+              </Group>
+              {!withOnHover && rightSide}
+
+              {withOnHover &&
+                rightSideWithHover({
+                  rightSide: rightSide,
+                  isHover: isHovered === link,
+                  tooltipLabel: tooltipLabel,
+                  popoverClasses: popoverClasses,
+                  popoverOpened: popoverOpened,
+                  setPopoverOpened: setPopoverOpened,
+                })}
+            </NavLink>
+          );
+        })}
     </div>
   );
 }
+
+const rightSideWithHover = ({
```

Comment:
This can be also potentially extracted to a separate component. I also feel like this is a unique use-case and we should keep this component code close to the icon, and we can pass the `hover` state from the parent.